### PR TITLE
Moving to nodal flag SLIP instead of (historical) variable IS_STRUCTURE

### LINKED ISSUE #214: coordinate_transformation_utilities.h
I'm working with the BDF scheme, and it is using the coordinate_transfromation_utilities.h. I see it is using the old flag IS_STRUCTURE varible. Is it worth it to adapt it to to use the STRUCTURE flag? What about the NORMAL variable used? Is it worth moving it to the Non-Historical DD.BB?

Need your comments: @jcotela, @rubenzorrilla @RiccardoRossi @KratosMultiphysics/fluid-dynamics

### LINKED ISSUE #1277: MPI flags synchronize
After a short discussion with @jcotela we realized that still not have any convention about the flag synchronization in MPI. As a preliminary idea, we think that we could add the next two methods:

- SynchronizeFlagAnd()
- SynchronizeFlagOr()

that perform the and and or boolean operations respectively. 

Any suggestion/idea about that?

### PR DESCRIPTION
**Pull request by jcotela**
As discussed in #214, #1277, I am replacing uses of `IS_STRUCTURE` with the nodal flag `SLIP` in the FluidDynamicsApplication and elsewhere. This should reduce the memory footprint of nodal data in fluid problems. I am creating a draft PR to keep track of the changes and so that people can start to test it.

I will continue to update the branch before it is ready for review, but this already includes the big behavior changes (rotation based on `SLIP` instead of `IS_STRUCTURE` and apply_slip_process no longer assigning `IS_STRUCTURE`).

This closes #214.

### CODE REVIEW

File: applications/FluidDynamicsApplication/custom_conditions/wall_condition_discontinuous.h [hunk 1]
```diff
@@ -308,7 +307,7 @@ class WallConditionDiscontinuous : public WallCondition<TDim,TNumNodes>
 //             noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize,LocalSize);
 //             noalias(rRightHandSideVector) = ZeroVector(LocalSize);
 //
-//             if(this->GetValue(IS_STRUCTURE) == 0.0 )
+//             if(this->Is(SLIP) == 0.0 )
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/custom_conditions/wall_condition_discontinuous.h]:
```suggestion
//             if (!this->Is(SLIP))
```

File: applications/FluidDynamicsApplication/python_scripts/vms_fractional_step_solver.py [hunk 2]
```diff
@@ -252,9 +251,9 @@ def Solve(self):
 # self.solver.ApplyFractionalVelocityFixity()
             (self.neighbour_search).Execute()
 # self.slip_conditions_initialized = False
-        if self.use_slip_conditions:
-            self.normal_util.CalculateOnSimplex(
-                self.model_part, self.domain_size, IS_STRUCTURE)
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/vms_fractional_step_solver.py:257]: I guess that you left this one for another PR. Am I wrong?

Comment (jcotela) [reply applications/FluidDynamicsApplication/python_scripts/vms_fractional_step_solver.py:257]: I will continue to work on this PR some more, but I do not know if we want to change this one. Current Fluid scripts use the version that computes the normals on an entire (sub)-modelpart, so we may want to leave this one for other applications (incompressible, ulf) which still use the variable.

I could also add an overload to `CalculateOnSimplex` that replaces the checks for IS_STRUCTURE with checks for a flag.

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/vms_fractional_step_solver.py:257]: I would do something like that. In any case this solver is not compatible with the json processes so we could rely that the users set the IS_STRUCTURE variable if they use it.

File: kratos/utilities/coordinate_transformation_utilities.h [hunk 3]
```diff
@@ -75,31 +75,14 @@ class CoordinateTransformationUtils {
...
 	 * @param NumRowsPerNode Number of matrix or vector rows associated to each node. Velocity DOFs are assumed to be the first mDomainSize rows in each block of rows.
-	 * @param rVariable Kratos variable used to flag nodes where local system contributions will be rotated. All nodes with rVariable != Zero will be rotated.
-	 * @param Zero The zero value for the variable.
+	 * @param rSelectionFlag All nodes where the flag given by this argument is set to true will be transformed to a rotated coordinate system.
 	 */
 	CoordinateTransformationUtils(const unsigned int DomainSize,
 			const unsigned int NumRowsPerNode,
-			const Variable<TValueType>& rVariable,
-			const TValueType Zero):
+			const Kratos::Flags& rSelectionFlag):
```
Comment (ddiezrod) [inline kratos/utilities/coordinate_transformation_utilities.h]: Maybe `const Kratos::Flags& rSelectionFlag = SLIP)`?

**Comment by jcotela** [comment]
I have updated all uses of `IS_STRUCTURE` in the context of the rotation tool that I could find. I believe that `IS_STRUCTURE` can now be removed from most FluidDynamicsApplication related code, at least for what concerns application of slip or wall conditions. @KratosMultiphysics/altair please let me know how you want to proceed with this PR. Should I just go ahead and merge once it is approved or do you need some time to adapt?

**Comment by ddiezrod** [comment]
I think you can go ahead and merge, we will take care of the changes in our side. Thanks @jcotela !

**Review by rubenzorrilla — APPROVED** [review]

**Comment by jcotela** [comment]
I'll go ahead and merge. However, since this PR affects quite a few parts I am not directly involved with, please do not hesitate to contact me if it turns out I broke something...

**Pull request merged**